src/main/java/de/bonndan/nivio/input/SourceReferencesResolver.java

```diff
@@ -47,7 +47,7 @@ public void resolve(final LandscapeDescription landscapeDescription) {
             }
 
             try {
-                landscapeDescription.addItems(factory.getDescriptions(ref, baseUrl));
+                factory.applyData(ref, baseUrl, landscapeDescription);
```

Comment:
variable name factory might be changed. It confuses with the spring bean factory.
 
``InputFormatHandler  inputFormatHandlerFactory;                                                              
inputFormatHandlerFactory.applyData(ref, baseUrl, landscapeDescription);``
This should be read able .i guess !!!